sylt-compiler/src/typechecker.rs

```diff
@@ -1133,8 +1136,20 @@ impl TypeChecker {
 
             ExpressionKind::Blob { blob, fields } => {
                 let blob_ty = self.type_assignable(ctx, blob)?;
+
+                let mut errors = Vec::new();
+
                 let (blob_name, blob_fields) = match self.find_type(blob_ty) {
                     Type::Blob(name, fields) => (name, fields),
+                    Type::Newblob(name, fields, _) => {
+                        errors.push(type_error!(
+                            self,
+                            span,
+                            TypeError::NewblobInstance { name: name.clone() }
+                        ));
```

Comment:
I've been bothered by how, when there are lots of errors, the most relevant one is often at the top, forcing you to scroll the terminal to find it. Python stacktraces often seem to know to put the most relevant error last. I know stacktraces and compilation errors are very different problems to solve, but maybe there is some inspiration we can gather anyway? For now, I think you're right.

Reply:
I find those errors a tad confusing, since I always read them the wrong way out of habit. But we can absolutely just reverse the list of errors before we print them out. It's a 1 line fix. If you want to do that, go for it!

Reply:
I decided to fail early instead.